Sources/SwiftDocC/Model/DocumentationNode.swift

```diff
@@ -421,6 +422,41 @@ public struct DocumentationNode {
             semantic.httpResponsesSectionVariants[.fallback] = HTTPResponsesSection(responses: responses)
         }
         
+        if let possibleValues = markupModel.discussionTags?.possibleValues, !possibleValues.isEmpty {
+            let validator = PossibleValuesSection.Validator(diagnosticEngine: engine)
+            guard let symbolAllowedValues = symbol?.mixins[SymbolGraph.Symbol.AllowedValues.mixinKey] as? SymbolGraph.Symbol.AllowedValues else {
+                possibleValues.forEach { 
+                    engine.emit(validator.makeExtraPossibleValueProblem($0, knownPossibleValues: []))
+                }
+                return semantic.possibleValuesSectionVariants[.fallback] = PossibleValuesSection(possibleValues: [])
+            }
+            
+            // Drop possible values defined in the markdown that are not defined
+            // in the SymbolGraph and emit a warning.
+            var knownPossibleValues = possibleValues.filter { possibleValue in
+                if symbolAllowedValues.value.contains(where: { String($0) == possibleValue.value }) {
+                    return true
+                }
+                // Record problem about extra possible value.
+                engine.emit(
+                    validator.makeExtraPossibleValueProblem(possibleValue, knownPossibleValues: Set(symbolAllowedValues.value.map { String($0) }))
+                )
+                return false
+            }
+            
+            // Add the symbol possible values that are not documented.
+            symbolAllowedValues.value.forEach { possibleValue in
+                if !knownPossibleValues.contains(where: { $0.value == String(possibleValue) }) {
+                    knownPossibleValues.append(
+                        PossibleValueTag(value: String(possibleValue), contents: [])
+                    )
+                }
+            }
```

Comment:
I generally try to avoid `forEach`. I find it to just be a less readable and less capable version of a regular for-loop.

Here, the pattern of an if-statement within a `forEach`:

```swift
someList.forEach { someValue in 
   if someCondition(someValue) {
       // do something
   }
}
```

is a less readable version of a for-loop with a where clause
```swift
for someValue in someList where someCondition(someValue) {
    // do something 
}
```

However, because the work that's done for each value is to add them to an array, this entire expression could be turned inside out so that a `knownPossibleValues.append(contentsOf:)` expression adds all the filtered values:
```swift
knownPossibleValues.append(contentsOf: symbolAllowedValues.value.compactMap { possibleValue in
    guard !knownPossibleValues.contains(where: { $0.value == String(possibleValue) }) else {
        return nil
    }
    return PossibleValueTag(value: String(possibleValue), contents: [])
})
```

There's a small inefficiency here (and in the original implementation) of recreating the possible value string for each check. Also, a `contains(where:)` check is generally slower than a `contains(_:)` check of a hashable value. 

Both of these could be solved by pre-computing the value names that each possible value is compared against:
```suggestion
            let knownPossibleValueNames = Set(knownPossibleValues.map(\.value))
            knownPossibleValues.append(contentsOf: symbolAllowedValues.value.compactMap { possibleValue in
                let possibleValueString = String(possibleValue)
                guard !knownPossibleValueNames.contains(possibleValueString) else {
                    return nil
                }
                return PossibleValueTag(value: possibleValueString, contents: [])
            })
```